src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -235,24 +238,74 @@ public function verifyEmail($verificationNonce)
         $secondFactorToVerify->verifyEmail();
     }
 
-    public function vetSecondFactor($registrationCode, $secondFactorIdentifier, $documentNumber, $identityVerified)
-    {
-        foreach ($this->verifiedSecondFactors as $secondFactor) {
-            if (!$secondFactor->wouldBeVettedBy(
-                $registrationCode,
-                $secondFactorIdentifier,
-                $documentNumber,
-                $identityVerified
-            )) {
-                continue;
-            }
+    public function vetSecondFactor(
+        IdentityApi $identity,
+        SecondFactorId $secondFactorId,
+        $secondFactorIdentifier,
+        $registrationCode,
+        $documentNumber,
+        $identityVerified
+    ) {
+        /** @var VettedSecondFactor|false $secondFactorWithHighestLoa */
+        $secondFactorWithHighestLoa = array_reduce(
+            $this->vettedSecondFactors->toArray(),
+            function (VettedSecondFactor $carry, VettedSecondFactor $item) {
+                return $carry->hasEqualOrHigherLoaComparedTo($item) ? $carry : $item;
+            },
+            $this->vettedSecondFactors->first()
+        );
```

Comment:
There is currently exactly one vetted second factor, rendering the `array_reduce` a bit redundant, not much to reduce if there is just one element - this could be masked by using the custom collection that actually returns just the one element it holds directly rather than using the reduce. Not that that has to happen, but would be fine to do.